### DISCUSSION TITLE
Deserializing third party types from array

### DISCUSSION THREAD

**Discussion by Charlie-83** [nlohmann/json#4220] [Q&A]
I am working with `glm` and so want to be able to deserialize things like `glm::vec3`. The way the documentation suggests would be something like
```json
{
  "x": 1,
  "y": 2,
  "z": 3
}
```

However, to reduce boiler plate, I would like the following instead:

```json
[1, 2, 3]
```

I have defined my appropriate `from_json` function but I now run into an error when running `json::parse(file)`. The error is about a size mismatch between vec3s. 

I assume that the parser is trying to interpret things as vec3s that shouldn't be interpreted as vec3s due to my form being more lenient than the suggested form. What I want is for `json::parse` to work in the "vanilla" way but then to be able to use my `from_json` function for extracting vec3s from specific places in the JSON later on. In particular, I use a lot of `std::vector<glm::vec3>` so would like to be able to deserialize these easily and without boilerplate.

What should I do here? Is there a way of doing this or is what I'm trying just a bad idea?

**Comment by nlohmann** [comment]
Can you please provide example code?

**Reply by Charlie-83** [reply]
Nvm, something completely unrelated was giving a warning that looked exactly like an error just before this and I assumed it had something to do with the error in the JSON stuff. The actual issue in the JSON stuff was a trailing comma. I'm very impressed by how well this library works
